diff --git a/libbb/platform.c b/libbb/platform.c
index 7913353e2..5997f1f32 100644
--- a/libbb/platform.c
+++ b/libbb/platform.c
@@ -8,6 +8,7 @@
  */
 #include "libbb.h"
 
+#if 0
 #ifndef HAVE_STRCHRNUL
 char* FAST_FUNC strchrnul(const char *s, int c)
 {
@@ -16,6 +17,7 @@ char* FAST_FUNC strchrnul(const char *s, int c)
 	return (char*)s;
 }
 #endif
+#endif
 
 #ifndef HAVE_USLEEP
 int FAST_FUNC usleep(unsigned usec)
@@ -225,3 +227,13 @@ int ttyname_r(int fd, char *buf, size_t buflen)
 	return 0;
 }
 #endif
+
+/* Android API 29 doesn't have explicit_bzero */
+#ifndef HAVE_EXPLICIT_BZERO
+void explicit_bzero(void *s, size_t n)
+{
+	memset(s, 0, n);
+	/* Prevent compiler from optimizing away the memset */
+	__asm__ __volatile__("" : : "r"(s) : "memory");
+}
+#endif
